pkg/apis/velero/v1alpha1/data_upload_types.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/vmware-tanzu/velero/pkg/apis/velero/shared"
+)
+
+// DataUploadSpec is the specification for a DataUpload.
+type DataUploadSpec struct {
+	// SnapshotType is the type of the snapshot to be backed up.
+	SnapshotType SnapshotType `json:"snapshotType"`
+
+	// If SnapshotType is CSI, CSISnapshot provides the information of the CSI snapshot.
+	// +optional
+	// +nullable
+	CSISnapshot *CSISnapshotSpec `json:"csiSnapshot"`
+
+	// SourcePVC is the name of the PVC which the snapshot is taken for.
+	SourcePVC string `json:"sourcePVC"`
```

Comment:
Is it possible that a user would want to backup a CSI Snapshot no longer related to a PVC in the cluster?
They may want to move snapshot to object storage and restore to a destination snapshot without a destination PVC.

Reply:
@kaovilai 
At least for the current CSI snapshot data movement workflow, it is not possible, because the DataUpload CR is created by the CSI plugin, and CSI plugin must take a PVC in order to create a CSI snapshot.
If this CRD is used in other scenarios, i.e., searching an existing VS and run a data movement, it is possible. For this case, `sourcePVC` could be set to "" as long as the underlying data mover doesn't use this value.

By the way, this discussion is more suitable to be in the design PR https://github.com/vmware-tanzu/velero/pull/5968, the design has include the details of the two CRDs

Reply:
Thanks! Still learning my way around :)